ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/ViewInterfaces.java

```diff
@@ -85,4 +87,48 @@
         }
         return names;
     }
+
+    static Set<Class<?>> getPotentialNotExposedViewInterfaces(Class<?> beanClass) {
+        Class<?>[] interfaces = beanClass.getInterfaces();
+        if (interfaces == null) {
+            return Collections.emptySet();
+        }
+        final Set<Class<?>> potentialBusinessInterfaces = new HashSet<Class<?>>();
+        for (Class<?> klass : interfaces) {
+            // EJB 3.1 FR 4.9.7 bullet 5.3
+            if (klass.equals(Serializable.class) ||
+                    klass.equals(Externalizable.class) ||
+                    klass.getName().startsWith("javax.ejb.") ||
+                    klass.getName().startsWith("groovy.lang.")) {
+                continue;
+            }
+            potentialBusinessInterfaces.add(klass);
+        }
+
+        // Check for other potential interfaces
+        Class superClass = beanClass.getSuperclass();
+        final Set<Class<?>> potentialNotExposedBusinessInterfaces = new HashSet<Class<?>>();
+
+        while(superClass!=null) {
+            Class<?>[] superClassInterfaces = superClass.getInterfaces();
+
+            if (superClassInterfaces != null) {
+                for (Class<?> klass : superClassInterfaces) {
+                    // EJB 3.1 FR 4.9.7 bullet 5.3
+                    if (klass.equals(Serializable.class) ||
+                            klass.equals(Externalizable.class) ||
+                            klass.getName().startsWith("javax.ejb.") ||
+                            klass.getName().startsWith("groovy.lang.") || potentialBusinessInterfaces.contains(klass)) {
+                        continue;
+                    }
+                    potentialNotExposedBusinessInterfaces.add(klass);
+                    EjbLogger.ROOT_LOGGER.potentialInterfaceExposure(klass.getCanonicalName());
+                }
+            }
+
+            superClass = superClass.getSuperclass();
+        }
+
+        return potentialNotExposedBusinessInterfaces;
```

Comment:
AFAICT this return value is never used, and the var is never read in this method, so the var can be dropped.

Reply:
Resolved.